train.py

```diff
@@ -1,234 +1,185 @@
+    os.makedirs(opt.output_dir, exist_ok=True)
     init_seeds()
     weights = 'weights' + os.sep
-    last = weights + 'last.pt'
-    best = weights + 'best.pt'
-    device = torch_utils.select_device(apex=mixed_precision)
+    last = os.path.join(opt.output_dir, 'last.pt')
+    best = os.path.join(opt.output_dir, 'best.pt')
+    device = torch_utils.select_device()
     multi_scale = opt.multi_scale
 
     if multi_scale:
-        img_sz_min = round(img_size / 32 / 1.5) + 1
-        img_sz_max = round(img_size / 32 * 1.5) - 1
-        img_size = img_sz_max * 32  # initiate with maximum multi_scale size
-        print('Using multi-scale %g - %g' % (img_sz_min * 32, img_size))
+        img_size_min = round(img_size / 32 / 1.5)
+        img_size_max = round(img_size / 32 * 1.5)
+        img_size = img_size_max * 32  # initiate with maximum multi_scale size
 
     # Configure run
-    data_dict = parse_data_cfg(data)
+    data_dict = parse_data_cfg(data_cfg)
     train_path = data_dict['train']
     nc = int(data_dict['classes'])  # number of classes
 
     # Initialize model
-    model = Darknet(cfg).to(device)
+    model = Darknet(cfg, img_size).to(device)
 
     # Optimizer
-    optimizer = optim.SGD(model.parameters(), lr=hyp['lr0'], momentum=hyp['momentum'], weight_decay=hyp['weight_decay'],
-                          nesterov=True)
-    # optimizer = AdaBound(model.parameters(), lr=hyp['lr0'], final_lr=0.1)
+    optimizer = optim.SGD(model.parameters(), lr=hyp['lr0'], momentum=hyp['momentum'], weight_decay=hyp['weight_decay'])
 
     cutoff = -1  # backbone reaches to cutoff layer
     start_epoch = 0
-    best_fitness = 0.
-    if opt.resume or opt.transfer:  # Load previously saved model
-        if opt.transfer:  # Transfer learning
-            nf = int(model.module_defs[model.yolo_layers[0] - 1]['filters'])  # yolo layer size (i.e. 255)
-            chkpt = torch.load(weights + 'yolov3-spp.pt', map_location=device)
-            model.load_state_dict({k: v for k, v in chkpt['model'].items() if v.numel() > 1 and v.shape[0] != 255},
-                                  strict=False)
-
-            for p in model.parameters():
-                p.requires_grad = True if p.shape[0] == nf else False
-
-        else:  # resume from last.pt
-            if opt.bucket:
-                os.system('gsutil cp gs://%s/last.pt %s' % (opt.bucket, last))  # download from bucket
-            chkpt = torch.load(last, map_location=device)  # load checkpoint
-            model.load_state_dict(chkpt['model'])
+    best_fitness = 0.0
+    best_results = None
+    best_efit = 0.0
+    if opt.resume: # resume from ckpt
+        chkpt = torch.load(opt.resume, map_location=device)  # load checkpoint
+        model.load_state_dict(chkpt['model'])
 
         if chkpt['optimizer'] is not None:
             optimizer.load_state_dict(chkpt['optimizer'])
             best_fitness = chkpt['best_fitness']
 
-        if chkpt.get('training_results') is not None:
```

Comment:
This is related to an older issue, if the key is missing it will raise an error so this code prevents that occuring.